src/xkit.cs

```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
 		NSTextRange GetAdjustedRange (NSTextRange textRange, bool forEditingTextSelection);
 	}
 
+	[TV (9,0), NoWatch, Mac (10,0), iOS (7,0)]
+	[BaseType (typeof (NSObject))]
+	interface NSTextList : NSCoding, NSCopying, NSSecureCoding {
+		[Export ("initWithMarkerFormat:options:")]
+#if NET
+		NativeHandle Constructor ([BindAs (typeof (NSTextListMarkerFormats))] NSString format, NSTextListOptions mask);
+#else
+		NativeHandle Constructor (string format, NSTextListOptions mask);
+#endif
+
+#if !NET
+		[Wrap ("this (format.GetConstant(), mask)")]
+		NativeHandle Constructor (NSTextListMarkerFormats format, NSTextListOptions mask);
+#endif
+
+#if NET
+		[BindAs (typeof (NSTextListMarkerFormats))] 
+#endif
+		[Export ("markerFormat")]
+#if NET
+		NSString MarkerFormat { get; }
+#else
+		string MarkerFormat { get; }
+#endif
+
+		[Export ("listOptions")]
+		NSTextListOptions ListOptions { get; }
+
+		[Export ("markerForItemNumber:")]
+		string GetMarker (nint itemNum);
+
+		[Export ("startingItemNumber")]
+		nint StartingItemNumber { get; set; }
+
+		[TV (16,0), NoWatch, Mac (13,0), iOS (16,0)]
+		[Export ("initWithMarkerFormat:options:startingItemNumber:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (string markerFormat, NSTextListOptions options, nint startingItemNumber);
+
+		[TV (16, 0), NoWatch, Mac (13, 0), iOS (16, 0)]
+		[Export ("ordered")]
+		bool Ordered { [Bind ("isOrdered")] get; }		
+	}
+
+	[TV (16,0), NoWatch, Mac (13,0), iOS (16,0)]
+	[BaseType (typeof(NSTextParagraph))]
+	interface NSTextListElement
+	{
+		[Export ("initWithAttributedString:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSAttributedString attributedString);
+
+		[Export ("initWithTextContentManager:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextContentManager textContentManager);
+
+		[Export ("initWithParentElement:textList:contents:markerAttributes:childElements:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextListElement parent, NSTextList textList, [NullAllowed] NSAttributedString contents, [NullAllowed] NSDictionary<NSString, NSObject> markerAttributes, [NullAllowed] NSTextListElement[] children);
```

Comment:
I would probably not use the generic dictionary here so we can create a better API in the future

```suggestion
		NativeHandle Constructor ([NullAllowed] NSTextListElement parent, NSTextList textList, [NullAllowed] NSAttributedString contents, [NullAllowed] NSDictionary markerAttributes, [NullAllowed] NSTextListElement[] children);
```